src/main/java/net/minecraftforge/client/gui/ScrollPanel.java

```diff
@@ -63,12 +72,41 @@ public ScrollPanel(Minecraft client, int width, int height, int top, int left)
         this.bottom = height + this.top;
         this.right = width + this.left;
         this.barLeft = this.left + this.width - barWidth;
+        this.border = border;
+        this.barWidth = barWidth;
+        this.bgColorFrom = bgColorFrom;
+        this.bgColorTo = bgColorTo;
+        this.barBgColor = barBgColor;
+        this.barColor = barColor;
+        this.barBorderColor = barBorderColor;
     }
 
     protected abstract int getContentHeight();
 
     protected void drawBackground() {}
 
+    protected void drawBackground(PoseStack matrix, Tesselator tess, float partialTicks)
```

Comment:
These two are badly named now. The one with no parameters is the first thing to be drawn while the second just renders the actual background. It would probably be good to mark one of these to have their name changed in 1.18 to better exemplify what they are supposed to be used for.

Reply:
I think the parameterless one is not useful at all so I would either 
- deprecate it or
- make a comment to rename it and add at least the PoseStack

Reply:
Unsure since it is drawn before the scissor is active. So it probably is better to comment and then rename instead.